CMakeLists.txt

```diff
@@ -109,22 +116,30 @@ if (NOT CMAKE_BUILD_TYPE OR ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
 endif()
 
 # Set preferred compiler warning flags
-set(WARN_FLAGS  -Wall
-                -Wextra
-                -Wcast-align
-                -Wunused
-                -Wconversion
-                -Wno-unknown-pragmas
-                -Wnon-virtual-dtor
-                -Wshadow)
-
-if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    set(WARN_FLAGS ${WARN_FLAGS}
-        -Wmisleading-indentation
-        -Wduplicated-cond
-        -Wduplicated-branches
-        -Wlogical-op
-        -Wnull-dereference)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+    set(WARN_FLAGS
+        -Wno-unknown-warning-option
+        -Wno-inconsistent-missing-override
+        -Wno-c++20-extensions
+        -Wno-deprecated-declarations)
+else ()
+    set(WARN_FLAGS
+        -Wall
```

Comment:
For Clang we'd also want all these warnings (or the equivalent clang warnings) in addition to what you had. Can you add those back?

Reply:
I was intimidated by -Werror but on closer look I think I can manage something.

Reply:
Yes, it took a long time to fix all the warnings, but it found things that were legitimate bugs. I'm sure turning on some clang errors will also help discover more.